Frontend/src/packages/editor/components/buttons/buttonHelpers.ts

```diff
@@ -0,0 +1,25 @@
+import { Editor, BaseEditor } from "slate";
+import { ReactEditor } from "slate-react";
+
+const toggleMark = (
+  editor: BaseEditor & ReactEditor,
+  format: "bold" | "italic" | "underline"
+): void => {
+  const isActive = isMarkActive(editor, format);
+
+  if (isActive) {
+    Editor.removeMark(editor, format);
+  } else {
+    Editor.addMark(editor, format, true);
+  }
+};
+
+const isMarkActive = (
+  editor: BaseEditor & ReactEditor,
+  format: "bold" | "italic" | "underline"
+): boolean => {
+  const marks = Editor.marks(editor);
+  return marks ? marks[format] === true : false;
```

Comment:
can you expand on what this line means maybe add a comment for the ternary operator

Reply:
Yep, comments added!